### PR TITLE
Feat: Theming brandless

### PR DESCRIPTION
**Pull request by renatamottam**
## What's the purpose of this pull request?
This PR tests and improves the Theming proposal. We extracted all BaseStore visual to a new theme called `soft-blue` and left global tokens more clean. We also added a new theme for testing purposes, it's called `midnight`.

Additionally, we updated Storybook doc with more sections and descriptions.

## How does it work?
Now you can create `themes`!
Just create a new .scss file on `src/styles/themes` and overwrite the tokens you want. 
We created two themes to use as examples: `soft-blue` and `midnight`.

## How to test it?
Checkout [our new Storybook](https://nextjs-store-storybook-git-feat-theming-brandless-faststore.vercel.app/), with a special attention to: [Getting Started](https://nextjs-store-storybook-git-feat-theming-brandless-faststore.vercel.app/?path=/docs/getting-started--page), [Customizing](https://nextjs-store-storybook-git-feat-theming-brandless-faststore.vercel.app/?path=/docs/theming-customizing--page) and [Typography](https://nextjs-store-storybook-git-feat-theming-brandless-faststore.vercel.app/?path=/docs/global-tokens-typography--page).

Screen Shot 2022-07-14 at 15 16 33

## Checklist

**Changelog**
- [x] Added an entry in the `CHANGELOG.md` at the beginning of its due section. [The latest version should comes first](https://keepachangelog.com/en/1.0.0/#:~:text=The%20latest%20version%20comes%20first.).
- [x] Added the PR number with the PR link at the entry in the `CHANGELOG.md`. E.g., *New items in the `pull_request_template.md` ([#4](https://github.com/vtex-sites/nextjs.store/pull/4))* 

**PR Description**
- [x] Added a label according to the PR goal - `Breaking change`, `Enhancement`, `Bug` or `Chore`.
- [x] Added the component, hook, or pathname in-between backticks (``) *- If applicable*. E.g., *`ComponentName` component*.

**Documentation**
- [x] PR description
- [x] Added to/Updated the Storybook - *if applicable*.
- [x] For documentation changes, ping @ carolinamenezes, @ PedroAntunesCosta or @ Mariana-Caetano to review and update.

### CODE REVIEW

**Comment by hellofanny** [comment]
I think it's okay, let's try to merge it after merging #142

File: src/components/product/ProductCard/product-card.module.scss [hunk 1]
```diff
@@ -80,6 +81,7 @@
       background-color: var(--fs-product-card-bkg-color-hover);
       border-color: var(--fs-product-card-border-color-hover);
       box-shadow: var(--fs-product-card-shadow-hover);
+      [data-product-card-image] img { transform: scale(var(--fs-product-card-img-scale-hover)); }
```
Comment (lucasfp13) [inline src/components/product/ProductCard/product-card.module.scss:84]: Wouldn't be the case to add `fs` here?

```suggestion
      [data-fs-product-card-image] img { transform: scale(var(--fs-product-card-img-scale-hover)); }
```

File: src/components/product/ProductCard/product-card.module.scss [hunk 2]
```diff
@@ -203,7 +215,8 @@
     }
 
     [data-product-card-image] {
```
Comment (lucasfp13) [inline src/components/product/ProductCard/product-card.module.scss:217]: the same here about the `fs`

**Review by lucasfp13 — APPROVED** [review]
⭐

File: src/styles/themes/custom-theme.scss [hunk 3]
```diff
@@ -0,0 +1,58 @@
+// ----------------------------------------------------------
```
Comment (eduardoformiga) [inline src/styles/themes/custom-theme.scss:1]: I wonder if this is the best place to have this file 🤔

File: src/pages/_app.tsx [hunk 4]
```diff
@@ -1,10 +1,14 @@
...
 import '../styles/global/tokens.scss'
 import '../styles/global/resets.scss'
 import '../styles/global/typography.scss'
 import '../styles/global/layout.scss'
 import '../styles/global/components.scss'
 
+// Replace these with your theme style file
+import '../styles/themes/midnight.scss'
+import '../styles/themes/soft-blue.scss'
+import '../styles/themes/custom-theme.scss'
```
Comment (eduardoformiga) [inline src/pages/_app.tsx:10]: I think we can just import the current theme.

**Review by eduardoformiga — COMMENTED** [review]
Amazing Job!! ✨✨✨

This PR is already merged but I left some discussions that we can deal with on other PRs.

**Pull request merged**